server/activate.go

```diff
@@ -52,18 +52,36 @@ func (p *Plugin) canSendDiagnostics() bool {
 	return enableDiagnostics != nil && *enableDiagnostics
 }
 
-func (p *Plugin) ensureBotExists() error {
+func (p *Plugin) ensureBotExists() *model.AppError {
 	// Attempt to find an existing bot
 	botUserIdBytes, err := p.API.KVGet(BOT_USER_KEY)
 	if err != nil {
 		return err
 	}
 
-	if botUserIdBytes == nil {
+	if botUserIdBytes != nil {
+		// Bot already exists
+		p.botUserId = string(botUserIdBytes)
+		return nil
+	}
+
+	var bot *model.Bot
+
+	if user, err := p.API.GetUserByUsername("surveybot"); err == nil && user != nil {
```

Comment:
Interesting. I wonder if we should have a library method that handles this and above.

Reply:
Yeah, it would be nice to have a "create or get" utility for bot accounts

Reply:
How would you identify the bot? By username?

Reply:
Oh, I think I misunderstood Chris's comment. I thought the "create or get" logic was what he was referring to, not that I got the bot as a regular user first.

I do think adding a `GetBotByUsername` method to the plugin API would be useful in cases like this, although it would be kind of nice to limit the number of ways to get them